Unlink account via bank sync page

Triscal opened the pull request:
First Pull request, so please take a close look and provide feedback. I would be very happy to learn. (Also if I'm overstepping in any of my edits please let me know, I want to respect the great work done by the team). :) 

Summary of changes:
- Added unlink button to bank sync edit modal. (Sidebar > Bank sync > Edit).
- Updated bank sync edit modal title to contain account name. 
- Replace user facing `bank-sync` with `bank sync` to have the same name everywhere. 

I built this feature because it was something I was looking for. There are currently two other ways to unlink accounts (from the account list over flow menu) and after loading the available accounts from the bank-sync providers (simplefin for example). But for me, when I was looking for the function, I thought the sidebar "bank sync" option would be able to help me do it, especially because it allows me to link accounts. 

I also updated the other times we refer to `bank sync` in the UI. The sidebar item is called `bank sync` and the link account modal has `bank sync` for gocardless but for a couple other spots we refer to it as `bank-sync`. Just for consistency I updated user / ui text to all be `bank sync`.

We already imported the account into the modal, so I only needed to import and call the unlink function. I also updated the modal's title to make it easier to tell which account you are editing. 

One potential edit / discussion point. Is that in the account menu you get a confirmation that you want to unlink, but in the "link account modal" you can do it with out confirmation. Since you are directly editing the bank sync settings, I didn't add a confirmation, but should there be one?

matt-fidd commented:
Hey, thanks for your work on this!

I think the button looks a little out of place at the top, what do you think to having it with the buttons on the bottom row? The red in the screenshot isn't great, I just used #f00 for speed, but there are some error colours defined in the themes you could use.

<img width="804" alt="image" src="https://github.com/user-attachments/assets/a246af68-6c98-44ba-accd-b01270e02b0d" />

I also think that there should be a confirmation box here, and you can include the "This page will close if account is unlinked." warning there.

matt-fidd commented on packages/desktop-client/src/components/banksync/EditSyncAccount.tsx:
```diff
@@ -200,10 +209,23 @@ export function EditSyncAccount({ account }: EditSyncAccountProps) {
         <>
           <ModalHeader
-            title={t('Account settings')}
+            title={account.name + t(' bank sync settings')}
```
It makes it easier to translate if the variable is included in the translation string.

```suggestion
            title={t('{{accountName}} bank sync settings', { accountName: account.name })}
```

matt-fidd reviewed (commented):
Couple of comments on the code, but it looks great overall. I agree with the standardisation of bank-sync vs bank sync too.

Triscal commented:
Thanks @matt-fidd! Appreciate you taking a look and the great feedback. I'll work on addressing your comments as soon as I can. 😄

Triscal commented on packages/desktop-client/src/components/banksync/EditSyncAccount.tsx:
```diff
@@ -200,10 +209,23 @@ export function EditSyncAccount({ account }: EditSyncAccountProps) {
+              onUnlink(close);
+            }}
+          >
+            <Trans>Unlink Account</Trans>
```
Fixed. :)

Triscal commented:
Okay, I think I the latest commit addresses all the feedback. 

I reused the `confirm-unlink-account` modal since it already had helpful text. I added a boolean check to add in the extra note about what is going to happen to the original modal, based off where the `confirm-unlink-account` modal is being opened from. In the same `confirm-unlink-account` modal I added <Trans> to unrelated string, just to get the whole modal translated. I realize this is not specifically related to the goal of this MR, but after seeing the text missing the translation tag I thought it may be useful to add it. 

I definitely agree that an excessively long account name could cause issues in the modal, I'm not sure deferring to the generic title is the correct solution. In other cases when it is too long we truncate it and an ellipsis (mainly the sidebar), IMO this is a great option because it still provides some info. Additionally in a world where some have specific titles it may appear that the generic title is actually an error / the settings not loading. Lastly from an accessibility standpoint, providing context is highly encouraged. 

Additionally the bank sync page has the same issue of a very long account name overrunning its container. I applied the same hard truncation after 25 characters. In both of these cases because the width of their container is fixed so it made sense IMO to add a simple truncation. Though I imagine and potentially saw other locations where a truncated account name could be helpful (in the unlink confirmation modal for example). But applying that logic more widely likely deserves its own discussion and potentially some centralized function to reduce duplication. 

I agree the bottom row is a better place for the button, I did some research and think I came up with a reasonable way to add spacing and push the unlink button to the left, but if there is a better way, I would be happy to hear it.

matt-fidd commented on packages/desktop-client/src/components/banksync/EditSyncAccount.tsx:
```diff
@@ -189,6 +213,9 @@ export function EditSyncAccount({ account }: EditSyncAccountProps) {
   };
 
+  const potentiallyTruncatedAccountName =
+    account.name.length > 25 ? account.name.slice(0, 25) + '...' : account.name;
```
```suggestion
    account.name.length > 30 ? account.name.slice(0, 30) + '...' : account.name;
```

matt-fidd commented on packages/desktop-client/src/components/banksync/AccountRow.tsx:
```diff
@@ -25,6 +25,11 @@ export const AccountRow = memo(
     });
 
+    const potentiallyTruncatedAccountName =
+      account.name.length > 25
+        ? account.name.slice(0, 25) + '...'
+        : account.name;
```
```suggestion
      account.name.length > 30
        ? account.name.slice(0, 30) + '...'
        : account.name;
```

matt-fidd reviewed (commented):
25 looks to cut it a bit too short, can we do 30 instead?

matt-fidd commented:
Great work, thanks for taking the feedback on board!

It won't let me comment over the unchanged lines to do a proper suggestion, but a flex wrapper would be cleaner for alignment of the buttons :)
```js
          <View
              style={{
                display: 'flex',
                flexDirection: 'row',
                alignItems: 'center',
                justifyContent: 'space-between',
                marginTop: 20,
              }}
          >
            <Button
              style={{ color: theme.errorText }}
              onPress={() => {
                onUnlink(close);
              }}
            >
              <Trans>Unlink account</Trans>
            </Button>

            <Stack direction="row" >
              <Button style={{ marginRight: 10 }} onPress={close}>
                <Trans>Cancel</Trans>
              </Button>
              <Button
                variant="primary"
                onPress={() => {
                  onSave(close);
                }}
              >
                <Trans>Save</Trans>
              </Button>
            </Stack>
          </View>
```

Triscal commented:
Ahh, nesting the rows. That does seem cleaner. Thanks for the suggestions

matt-fidd reviewed (approved):
Looks great, thank you! I really like the way this is implemented.
